frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -891,9 +890,25 @@ export default ({
         if (!value) throw new Error('Unable to decrypt message')
 
         const isMessageAddedOrDeleted = (message: GIMessage) => {
-          if (![GIMessage.OP_ACTION_ENCRYPTED, GIMessage.OP_ACTION_UNENCRYPTED].includes(message.opType())) return {}
+          const allowedActionType = [GIMessage.OP_ACTION_ENCRYPTED, GIMessage.OP_ACTION_UNENCRYPTED]
+          let action = null
+          if (message.opType() === GIMessage.OP_ATOMIC) {
+            const opTypes = value.map(([type, value]) => type)
+            for (let i = 0; i < opTypes.length; i++) {
+              if (allowedActionType.includes(opTypes[i])) {
+                // NOTE: choose the first action with allowed opType
+                //       which also would be the only child action of OP_ATOMIC message
+                const v = value[i][1].valueOf()
+                action = !Object.getPrototypeOf(v)?._isSignedData ? v.action : v?.valueOf()?.action
```

Comment:
The `_isSignedData` isn't a stable API and is also weird. I'd change that to `.valueOf().valueOf()`, which always works, and add a comment explaining why.